backend/api/views.py

```diff
@@ -1,77 +1,74 @@
-                                        password=request.data['password'])
-    return Response(f'User with id {user.id} and email {user.email} was created', status=status.HTTP_201_CREATED)
 
+class RegisterUser(APIView):
+    permission_classes = (AllowAny, )
 
-@api_view(['POST'])
-def authenticateUser(request):
-    # Note that username and email are the same
-    user = authenticate(username=request.data['email'], password=request.data['password'])
-    if user is not None:
-        # TODO login(request,user)
-        # redirect to a success page
-        return Response(f'Authenticated user {user.email}', status=status.HTTP_200_OK)
-    else:
-        return Response('Invalid login', status=status.HTTP_401_UNAUTHORIZED)
+    def post(self, request):
+            # Note that username and email are the same
+            user = FiboUser.objects.create_user(username=request.data['email'], email=request.data['email'],
+                                                password=request.data['password'])
+            return Response(f'User with id {user.id} and email {user.email} was created', status=status.HTTP_201_CREATED)
 
 
-@api_view(['GET'])
-def getRoutes(request):
-    routes = [
-        {
-            'Endpoint': '/users/register',
-            'method': 'POST',
-            'body': {'email': '', 'password': ''},
-            'description': 'Register Endpoint for a new user!'
-        },
-        {
-            'Endpoint': '/users/authenticate',
-            'method': 'POST',
-            'body': {'email': '', 'password': ''},
-            'description': 'Authenticates user with given email and password'
-        },
-        {
-            'Endpoint': '/users/<id>/delete',
-            'method': 'DELETE',
-            'body': None,
-            'description': 'Deletes user with given id'
-        },
-        {
-            'Endpoint': '/users/<id>',
-            'method': 'GET',
-            'body': None,
-            'description': 'Returns user with given id'
-        },
-        {
-            'Endpoint': '/users/<id>/update',
-            'method': 'PUT',
-            'body': {'email': '', 'newPassword': '', 'oldPassword': ''},
-            'description': 'Not implemented yet. Updates user with data sent in put request'
-        },
-    ]
-    return Response(routes)
+class GetRoutes(APIView):
+    permission_classes = (IsAuthenticated, )
+
+    def get(self):
+        routes = [
+            {
+                'Endpoint': '/users/register',
+                'method': 'POST',
+                'body': {'email': '', 'password': ''},
+                'description': 'Register Endpoint for a new user!'
+            },
+            {
+                'Endpoint': '/users/authenticate',
+                'method': 'POST',
+                'body': {'email': '', 'password': ''},
+                'description': 'Authenticates user with given email and password'
```

Comment:
Please change the documentation here so that it will be easily understandable what parameters need to be send in each http request.

Reply:
Changed it. I dont really know what else to write, so if you're missing something, feel free to tell me. :)

Reply:
I'm not sure about what part of the api requires the token to be sent with and what part requires the password. The 'body' is probably old in either the /authenticate documentation or the /login part of this documentation so maybe you could have a look again and change the 'body' and the 'description' according to the new api views?